cumulus/polkadot-parachain/src/service/mod.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+mod consensus;
+mod new_partial;
+mod rpc_extensions;
+mod start_node_impl;
+mod start_nodes;
+
+use cumulus_client_consensus_aura::collators::lookahead::{self as aura, Params as AuraParams};
+use cumulus_client_consensus_common::ParachainBlockImport as TParachainBlockImport;
+use sc_executor::WasmExecutor;
+use sc_service::{PartialComponents, TFullBackend, TFullClient};
+use sc_telemetry::{Telemetry, TelemetryWorkerHandle};
+use std::sync::Arc;
+
+// Exports from the service module
+
+pub use consensus::{
+	lookahead_aura_consensus::{build_aura_import_queue, start_lookahead_aura_consensus},
+	relay_chain_consensus::build_relay_to_aura_import_queue,
+};
+pub use new_partial::new_partial;
+pub use rpc_extensions::{build_contracts_rpc_extensions, build_parachain_rpc_extensions};
+pub use start_node_impl::start_node_impl;
+pub use start_nodes::{
+	asset_hub_lookahead::start_asset_hub_lookahead_node,
+	basic_lookahead::start_basic_lookahead_node,
+	generic_aura_lookahead::start_generic_aura_lookahead_node,
+	rococo_contracts::start_contracts_rococo_node,
+	rococo_parachain::start_rococo_parachain_node,
+	shell::{build_shell_import_queue, start_shell_node},
+};
+
+/// Module with common types.
+///
+/// The idea is to help downstream implementors who don't rely on `parachains_common` for types.
+/// Proxying the imports paths via this module makes it possible to only adjust imports here, as
+/// opposed to replacing `parachains_common` imports in every file.
+pub mod common_types {
```

Comment:
I don't think this makes sense. `polkadot-parachain` crate is just code that is necessary to build the `polkadot-parachain` binary. It has no external facing API, so there are no downstream implementors using this. We should continue using `parachains_common` here.

Reply:
Maybe I badly formulated here. The reasoning here was that if a parachain team updates the polkadot version, and they want to integrate the changes by comparing the source files, they need to change the files in more places instead of just here.

Moonbeam for example would have to change the `parachains_common::AccountId` import in about four places now if I remember correctly, but I don't really like this re-export thing either, but I think it is better than the status quo.